components/butterfly/src/rumor/service.rs

```diff
@@ -80,16 +81,23 @@ impl DerefMut for Service {
 
 impl Service {
     /// Creates a new Service.
-    pub fn new<S1, S2, S3>(member_id: S1,
-                           service_group: ServiceGroup,
-                           hostname: S2,
-                           ip: S3,
-                           exposes: Vec<u32>)
-                           -> Self
+    pub fn new<S1, S2, S3, S4, S5>(member_id: S1,
+                                   package: &S4,
+                                   group: S5,
+                                   organization: Option<String>,
+                                   hostname: S2,
+                                   ip: S3,
+                                   exposes: Vec<u32>)
+                                   -> Self
         where S1: Into<String>,
               S2: Into<String>,
-              S3: Into<String>
+              S3: Into<String>,
+              S4: Identifiable,
+              S5: Into<String>
     {
+        assert!(package.fully_qualified(),
```

Comment:
I don't think there is anything to do about it right this second, but I have a lurking fear that we're taking information from the network and possibly causing ourselves to panic. I think that's not the case here, but I can feel it in Butterfly.